### PR TITLE
extend root partition on first boot

### PR DESCRIPTION
**Pull request by samariezas**
Based on [this article](https://gijs-de-jong.nl/posts/raspberry-pi-file-system-resize-on-first-boot/) of [Raspberry Pi disk expansion script](https://github.com/RPi-Distro/raspi-config/blob/master/usr/lib/raspi-config/init_resize.sh).

Outline:
1. Replace init process with shell script.
2. On first boot, remove init= kernel option from grub config. This way script runs only once.
3. Expand GPT, partition and resize2fs.

Tested in QEMU and on bare metal with only USB stick plugged in, without any other disks attached.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@ QEMU ?= qemu-system-x86_64 -enable-kvm -cpu host
 iso: $(BASENAME).iso
 
 $(BASENAME).iso:
-	lb config
+	lb config --archive-areas "main contrib non-free non-free-firmware"
```
Comment (aursulis) [inline Makefile:11]: Is this change required? I've already made a similar one by adding `non-free-firmware` to `auto/config` here: https://github.com/aursulis/liox/blob/master/auto/config#L26

Comment (samariezas) [reply Makefile:11]: For some reason, I still had issues while building with the Makefile. Possibly differences between distros? I'm 90% sure this script was being run, as I saw the LIOX_ARCH error message when I forgot to set variable. I will look into this tomorrow

Comment (samariezas) [reply Makefile:11]: Tested again, still fails.
image
